### PR TITLE
Implement GCC's switch-case ranges

### PR DESCRIPTION
**Pull request by tautschnig**

### CODE REVIEW

File: src/goto-programs/goto_convert.cpp [hunk 1]
```diff
@@ -429,22 +430,32 @@ void goto_convertt::convert_gcc_switch_case_range(
...
   }
 
-  // TODO
-  exprt::operandst &case_op_dest=cases_entry->second->second;
-  case_op_dest.push_back(code.case_op());
-  #endif
+  const auto lb = numeric_cast<mp_integer>(code.op0());
+  const auto ub = numeric_cast<mp_integer>(code.op1());
+
+  if(!lb.has_value() || !ub.has_value())
```
Comment (smowton) [inline src/goto-programs/goto_convert.cpp]: Perhaps check `lb <= ub` here too?

File: src/goto-programs/goto_convert.cpp [hunk 2]
```diff
@@ -429,22 +430,32 @@ void goto_convertt::convert_gcc_switch_case_range(
...
+  {
+    error().source_location = code.find_source_location();
+    error() << "GCC's switch-case-range statement requires integer bounds"
+            << eom;
```
Comment (kroening) [inline src/goto-programs/goto_convert.cpp]: requires _constant_ integer bounds?

**Review by kroening — APPROVED** [review]

File: src/goto-programs/goto_convert.cpp (hunk 1 above)
Comment (tautschnig) [reply src/goto-programs/goto_convert.cpp]: Added a warning and included that case in the regression test.

File: src/goto-programs/goto_convert.cpp (hunk 2 above)
Comment (tautschnig) [reply src/goto-programs/goto_convert.cpp]: Done.

**Review by smowton — APPROVED** [review]

**Pull request merged**